src/scss/objects/_form-elements.scss

```diff
@@ -0,0 +1,241 @@
+/**
+ * Custom Checkboxes and radios
+ *
+ * --------------------------------
+ *
+ * Checkbox button example HTML
+ *
+ *    <label class="c-form-label" for="checkbox1">
+ *        <input class="c-form-input" type="checkbox" id="checkbox1" name="myCheckbox" value="A value"/>
+ *        <span class="c-form-indicator c-form-indicator--checkbox"></span>
+ *        My Checkbox Label Text
+ *    </label>
+ *
+ * -------------------------------
+ *
+ * Radio button example HTML
+ *
+ *    <label class="c-form-label" for="radio1">
+ *        <input class="c-form-input" type="radio" id="radio1" name="myRadioBtn" value="A value"/>
+ *        <span class="c-form-indicator c-form-indicator--radio"></span>
+ *        My Radio Button Label Text
+ *     </label>
+ *
+ * ================================= */
+ $input-bgColor                  : $white !default;
+ $input-bgColor-disabled         : $color-disabled;
+ $input-borderColor              : $color-border;
+ $input-color             : $blue !default;
+ $input-background-color  : $color-bg !default;
+ $input-hover-color       : $blue--dark !default;
+ $input-borderColor-hover : $blue !default;
+ $input-margin-left       : 0 !default;
+ $input-padding-left      : spacing(x4) !default;
+ $input-width--wide       : 20px !default;
+ $input-width             : 24px !default;
+ $input-height--wide      : 20px !default;
+ $input-height            : 24px !default;
+ $input-border-width      : 1px;
+
+
+ .o-form-label {
+     position: relative;
+     display: block;
+     cursor: pointer;
+     user-select: none;
+     padding: spacing();
+     padding-left: $input-padding-left;
+
+     &:hover .o-form-indicator {
+         border-color: $input-borderColor-hover;
+     }
+
+     &:hover .o-form-input:disabled ~ .o-form-indicator {
+         border-color: $input-borderColor;
+         cursor: not-allowed;
+     }
+
+     .o-form-input:disabled ~ .o-form-indicator {
+        background-color: $input-bgColor-disabled;
+     }
+ }
+
+ // Give inline elements some space to the right
+ .o-form-label--inline {
+     display: inline-block;
+     padding-right: 2em;
+ }
+
+     // The hidden input
+     .o-form-input {
```

Comment:
As this isn't called `control-input` I'd probably call it something a little more specific to checkboxes and radios, else it may clash with other text inputs when we get around to styling those.

Not sure what best to name it though.  Maybe `o-form-control` or something like that? (as that then represents both input types)